livedata/src/main/java/io/getstream/chat/android/livedata/repository/UserRepository.kt

```diff
@@ -5,81 +5,89 @@ import io.getstream.chat.android.client.models.User
 import io.getstream.chat.android.livedata.dao.UserDao
 import io.getstream.chat.android.livedata.entity.UserEntity
 
-class UserRepository(
-    var userDao: UserDao,
-    var cacheSize: Int = 100,
-    var currentUser: User?
+internal class UserRepository(
+    private val userDao: UserDao,
+    private val currentUser: User?,
+    cacheSize: Int = 100
 ) {
     // the user cache is simple, just keeps the last 100 users in memory
-    var userCache = LruCache<String, UserEntity>(cacheSize)
+    var userCache = LruCache<String, User>(cacheSize)
 
-    suspend fun insert(userEntities: List<UserEntity>) {
-        if (userEntities.isEmpty()) return
-        cacheUserEntities(userEntities)
-        userDao.insertMany(userEntities)
+    suspend fun insert(users: List<User>) {
+        if (users.isEmpty()) return
+        cacheUsers(users)
+        userDao.insertMany(users.map(::toEntity))
     }
 
-    private fun cacheUserEntities(userEntities: List<UserEntity>) {
-        for (userEntity in userEntities) {
+    private fun cacheUsers(users: List<User>) {
+        for (userEntity in users) {
             userCache.put(userEntity.id, userEntity)
         }
     }
 
-    suspend fun insertManyUsers(users: List<User>) {
-        val userEntities = users.map { UserEntity(it) }
-        insert(userEntities)
-    }
-
     suspend fun insertUser(user: User) {
-        val userEntity = UserEntity(user)
-        insert(listOf(userEntity))
+        userDao.insert(toEntity(user))
     }
 
     suspend fun insertMe(user: User) {
-        val userEntity = UserEntity(user)
-        userEntity.originalId = user.id
-        userEntity.id = "me"
-        insert(listOf(userEntity))
+        val userEntity = toEntity(user).copy(originalId = user.id, id = ME_ID)
```

Comment:
Could we move it into `toEntity()` method?
We can assign `originalId` as `user.id` to all entities, and only edit the `id` when it is the current user.
It would simplify the `toModel()` method

Reply:
Yes, good idea. Implemented it